### PR TITLE
Fix regression with DiffableMaps and duplicate agglomerate skeletons

### PR DESCRIPTION
**Pull request by philippotto**
This PR fixes an issue with with WK loading duplicate tree when enabling agglomerate skeletons. This bug is a regression from #8596 .

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Load a agglomerate skeleton
- Verify that only one tree was loaded
- Do any operation (spllit, rename) on the agglomerate skeleton without issue
- also see added tests

### Issues:
- fixes https://scm.slack.com/archives/C02H5T8Q08P/p1748263747190599

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: frontend/javascripts/test/reducers/skeletontracing_reducer.spec.ts [hunk 1]
```diff
@@ -1784,4 +1791,48 @@ describe("SkeletonTracing", () => {
...
+        new MutableTreeMap([[4, { ...initalTreeOne, treeId: 4 }]]),
+        [],
+        treeIdCallback,
+      ),
+    ]);
+
+    // The original tree id 4 should have survived.
+    expect(ids).toEqual([4]);
+    expect(state.annotation.skeleton?.trees.size()).toBe(1);
+    expect(state.annotation.skeleton?.trees.getNullable(4)).toBeDefined();
```
Comment (hotzenklotz) [inline frontend/javascripts/test/reducers/skeletontracing_reducer.spec.ts:1816]: Does it make sense to also test that the newly added tree has tree id 5(?) now?
